src/shell/commands/bulk_load.cpp

```diff
@@ -0,0 +1,212 @@
+        fprintf(stderr, "app_name should not be empty\n");
+        return false;
+    }
+    if (cluster_name.empty()) {
+        fprintf(stderr, "cluster_name should not be empty\n");
+        return false;
+    }
+
+    if (file_provider_type.empty()) {
+        fprintf(stderr, "file_provider_type should not be empty\n");
+        return false;
+    }
+
+    auto err_resp = sc->ddl_client->start_bulk_load(app_name, cluster_name, file_provider_type);
+    dsn::error_s err = err_resp.get_error();
+    std::string hint_msg;
+    if (err.is_ok()) {
+        err = dsn::error_s::make(err_resp.get_value().err);
+        hint_msg = err_resp.get_value().hint_msg;
+    }
+    if (!err.is_ok()) {
+        fmt::print(stderr, "start bulk load failed, error={} [hint:\"{}\"]\n", err, hint_msg);
+    } else {
+        fmt::print(stdout, "start bulk load succeed\n");
+    }
+
+    return true;
+}
+
+bool pause_bulk_load(command_executor *e, shell_context *sc, arguments args)
+{
+    static struct option long_options[] = {{"app_name", required_argument, 0, 'a'}, {0, 0, 0, 0}};
+    std::string app_name;
+
+    optind = 0;
+    while (true) {
+        int option_index = 0;
+        int c;
+        c = getopt_long(args.argc, args.argv, "a:", long_options, &option_index);
+        if (c == -1)
+            break;
+        switch (c) {
+        case 'a':
+            app_name = optarg;
+            break;
+        default:
+            return false;
+        }
+    }
+
+    if (app_name.empty()) {
+        fprintf(stderr, "app_name should not be empty\n");
+        return false;
+    }
+
+    auto err_resp = sc->ddl_client->control_bulk_load(
+        app_name, dsn::replication::bulk_load_control_type::BLC_PAUSE);
+    dsn::error_s err = err_resp.get_error();
+    std::string hint_msg;
+    if (err.is_ok()) {
+        err = dsn::error_s::make(err_resp.get_value().err);
+        hint_msg = err_resp.get_value().hint_msg;
+    }
+    if (!err.is_ok()) {
+        fmt::print(stderr, "pause bulk load failed, error={} [hint:\"{}\"]\n", err, hint_msg);
+    } else {
+        fmt::print(stdout, "pause bulk load succeed\n");
+    }
+
+    return true;
+}
+
+bool restart_bulk_load(command_executor *e, shell_context *sc, arguments args)
+{
+    static struct option long_options[] = {{"app_name", required_argument, 0, 'a'}, {0, 0, 0, 0}};
+    std::string app_name;
+
+    optind = 0;
+    while (true) {
+        int option_index = 0;
+        int c;
+        c = getopt_long(args.argc, args.argv, "a:", long_options, &option_index);
+        if (c == -1)
+            break;
+        switch (c) {
+        case 'a':
+            app_name = optarg;
+            break;
+        default:
+            return false;
+        }
+    }
+
+    if (app_name.empty()) {
+        fprintf(stderr, "app_name should not be empty\n");
+        return false;
+    }
+
+    auto err_resp = sc->ddl_client->control_bulk_load(
+        app_name, dsn::replication::bulk_load_control_type::BLC_RESTART);
```

Comment:
Almost same with pause_bulk_load except `bulk_load_control_type `, you can do some encapsulation and reuse some code.

Reply:
done